src/vm/eventpipe.cpp

```diff
@@ -107,7 +108,7 @@ void EventPipe::Shutdown()
         NOTHROW;
         GC_TRIGGERS;
         MODE_ANY;
-        PRECONDITION(s_tracingInitialized);
+        PRECONDITION(s_state != EventPipeState::ShuttingDown);
```

Comment:
I thought we had some race conditions that can (legally) call Shutdown concurrently? For example someone uses ctrl-C just as the a console app was exiting gracefully.

Reply:
`EventPipe::Shutdown` is only called within the graceful Clr shutdown only once :)